agentlib/core/agent.py

```diff
@@ -51,22 +58,37 @@ class AgentConfig(BaseModel):
         default=1000,
         ge=-1,
         description="Maximal number of waiting items in data-broker queues. "
-                    "Set to -1 for infinity"
+        "Set to -1 for infinity",
     )
 
     @field_validator("modules")
     @classmethod
-    def check_modules(cls, modules: List):
-        """Validator to ensure all modules are in dict-format."""
+    def check_modules(cls, modules: Union[List, Dict]):
+        """Validator to ensure all modules are in dict-format and include 'module_id'."""
         modules_loaded = []
-        for module in modules:
-            if isinstance(module, (str, Path)):
-                if Path(module).exists():
-                    with open(module, "r") as f:
-                        module = json.load(f)
-                else:
-                    module = json.loads(module)
-            modules_loaded.append(module)
+        if isinstance(modules, dict):
+            for module_id, module in modules.items():
+                if isinstance(module, (str, Path)):
```

Comment:
So we just support both and don't do deprication warnings? If so, you can ignore my comments below.